File: src/main/java/net/lomeli/trophyslots/compat/PlayerInvMod.java
package net.lomeli.trophyslots.compat;

import java.util.List;

import net.minecraft.client.gui.inventory.GuiContainer;
import net.minecraft.client.gui.inventory.GuiInventory;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.inventory.ContainerPlayer;
import net.minecraft.inventory.Slot;
import net.minecraft.inventory.SlotCrafting;

import net.lomeli.trophyslots.TrophySlots;
import net.lomeli.trophyslots.client.slots.SlotLocked;

public class PlayerInvMod implements ICompatModule {
    @Override
    public void replaceSlots(GuiContainer gui, EntityPlayer player) {
        ContainerPlayer containerPlayer = new ContainerPlayer(player.inventory, !player.worldObj.isRemote, player);
        List slotList = containerPlayer.inventorySlots;
        if (slotList != null) {
            for (int i = 5; i < slotList.size(); i++) {
                Slot slot = containerPlayer.getSlot(i);
                if (slot != null && !(slot instanceof SlotCrafting)) {
                    if (slot.inventory != containerPlayer.craftMatrix && slot.isHere(player.inventory, slot.getSlotIndex()) && !TrophySlots.proxy.slotUnlocked(slot.getSlotIndex()))
                        containerPlayer.inventorySlots.set(i, new SlotLocked(player.inventory, slot.getSlotIndex(), slot.xDisplayPosition, slot.yDisplayPosition));
                }
            }
        }
        gui.inventorySlots = containerPlayer;
    }

    @Override
    public boolean isCompatibleGui(GuiContainer gui) {
        return gui instanceof GuiInventory;
    }
}

File: src/main/kotlin/net/lomeli/trophyslots/compat/PlayerInvMod.kt
package net.lomeli.trophyslots.compat

import net.lomeli.trophyslots.capabilities.slots.SlotManager
import net.lomeli.trophyslots.client.slots.SlotLocked
import net.minecraft.client.gui.inventory.GuiContainer
import net.minecraft.client.gui.inventory.GuiInventory
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.inventory.ContainerPlayer
import net.minecraft.inventory.SlotCrafting

object PlayerInvMod : ICompatModule {
    override fun replaceSlots(container: GuiContainer, player: EntityPlayer) {
        val containerPlayer = ContainerPlayer(player.inventory, !player.world.isRemote, player)
        val slotList = containerPlayer.inventorySlots
        val slotInfo = SlotManager.getPlayerSlotInfo(player)!!
        if (slotList != null) {
            for (i in slotList.indices) {
                val slot = containerPlayer.getSlot(i)
                if (slot != null && slot !is SlotCrafting) {
                    if (slot.inventory !== containerPlayer.craftMatrix && slot.isHere(player.inventory, slot.slotIndex) && !slotInfo.slotUnlocked(slot.slotIndex))
                        containerPlayer.inventorySlots.set(i, SlotLocked(player.inventory, slot.slotIndex, slot.xPos, slot.yPos))
                }
            }
        }
        container.inventorySlots = containerPlayer
    }

    override fun isCompatibleGui(gui: GuiContainer): Boolean = gui is GuiInventory
}
